src/python/pants/net/http/fetcher.py

```diff
@@ -126,6 +128,7 @@ def __init__(self, digest=None):
       self._checksum = None
 
     def recv_chunk(self, data):
+      data = [ensure_binary(v) for v in data] if isinstance(data, list) else ensure_binary(data)
```

Comment:
Doing a check for list to avoid converting a string into individual letters, which was happening with one test case.